### ISSUE TITLE
parser: optional type annotation on `VariableDeclarator` does not produce a syntax error

### ISSUE THREAD

**Issue by DonIsaac** [oxc-project/oxc#5955]
This snippet should produce a syntax error, but it does not.

```ts
const x?: number = 1;
```
